pkg/reconcilermanager/controllers/reposync_controller_test.go

```diff
@@ -2402,40 +2417,58 @@ func TestRepoSyncSpecValidation(t *testing.T) {
 		t.Error(err)
 	}
 
-	// verify redundant source specification
-	rs.Spec.SourceType = string(v1beta1.GitSource)
-	rs.Spec.Git = &v1beta1.Git{}
-	rs.Spec.Oci = &v1beta1.Oci{}
+	// verify missing Helm repo
+	rs.Spec.SourceType = string(v1beta1.HelmSource)
+	rs.Spec.Oci = nil
+	rs.Spec.Helm = &v1beta1.Helm{}
 	if err := fakeClient.Update(ctx, rs); err != nil {
-		t.Fatalf("failed to update the root sync request, got error: %v", err)
+		t.Fatalf("failed to update the repo sync request, got error: %v", err)
 	}
 	if _, err := testReconciler.Reconcile(ctx, reqNamespacedName); err != nil {
 		t.Fatalf("unexpected reconciliation error, got error: %q, want error: nil", err)
 	}
 	wantRs.Spec = rs.Spec
-	reposync.SetStalled(wantRs, "Validation", validate.RedundantOciSpec(rs))
+	reposync.SetStalled(wantRs, "Validation", validate.MissingHelmRepo(rs))
 	if err := validateRepoSyncStatus(wantRs, fakeClient); err != nil {
 		t.Error(err)
 	}
 
-	rs.Spec.SourceType = string(v1beta1.OciSource)
-	rs.Spec.Git = &v1beta1.Git{}
-	rs.Spec.Oci = &v1beta1.Oci{}
+	// verify missing Helm chart
+	rs.Spec.SourceType = string(v1beta1.HelmSource)
+	rs.Spec.Helm = &v1beta1.Helm{Repo: helmRepo}
+	if err := fakeClient.Update(ctx, rs); err != nil {
+		t.Fatalf("failed to update the repo sync request, got error: %v", err)
+	}
+	if _, err := testReconciler.Reconcile(ctx, reqNamespacedName); err != nil {
+		t.Fatalf("unexpected reconciliation error, got error: %q, want error: nil", err)
+	}
+	wantRs.Spec = rs.Spec
+	reposync.SetStalled(wantRs, "Validation", validate.MissingHelmChart(rs))
+	if err := validateRepoSyncStatus(wantRs, fakeClient); err != nil {
+		t.Error(err)
+	}
+
+	// verify invalid Helm Auth
+	// Mock out parseDeployment for testing.
+	parseDeployment = parsedDeployment
```

Comment:
This is not needed as there is no change to the mock Deployment. You may want to delete https://github.com/GoogleContainerTools/kpt-config-sync/pull/50/files#diff-4e452cd48e11abd5a08fa09505577300d00a536c7c57d7ebc492a9b166be3f55R2405 as well.

Reply:
Done